tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h

```diff
@@ -164,17 +111,13 @@ inline void calculate_sine()
     // SFPU microcode
     for (int d = 0; d < ITERATIONS; d++)
     {
-        vFloat v = dst_reg[0];
-        v = 0.318309886183791f*v; // *1/pi to get number of pi rads.
+        vFloat v = dst_reg[0] * FRAC_1_PI;
         vInt whole_v = float_to_int16(v);
-        vFloat whole_v_float = int32_to_float(whole_v, 0);
-        v = v - whole_v_float;
-        v *= 3.141592653589793f; // fractional * pi to get it in [-pi:pi]
-        v = sfpu_sine_maclaurin_series<APPROXIMATION_MODE>(v);
-        whole_v = whole_v & 0x1;
-        v_if(whole_v != 0) {
-            // odd so flip the sign
-            v *= -1;
+        v -= int32_to_float(whole_v, 0);
```

Comment:
We don't approximate in [-π, π] but in [-π/2, π/2].  `float_to_int16` rounds to the nearest integer rather than truncates.